fastkml/data.py

```diff
@@ -1,20 +1,32 @@
+from typing import Dict
 from typing import List
 from typing import Optional
 from typing import Tuple
 from typing import Union
 
 from typing_extensions import TypedDict
 
-import fastkml.config as config
 from fastkml.base import _BaseObject
 from fastkml.base import _XMLObject
+from fastkml.config import etree  # type: ignore[attr-defined]
 from fastkml.types import Element
 
 
 class SimpleField(TypedDict):
     name: str
     type: str
-    displayName: str  # noqa: N815
+    displayName: Optional[str]  # noqa: N815
+
+
+SimpleFields = List[Dict[str, Union[str, str]]]
+SimpleFieldsInput = Optional[
```

Comment:
This type annotation looks too complex to me. Please try to define smaller subtypes and name them according to their use.
Also consider using [`@overload`](https://adamj.eu/tech/2021/05/29/python-type-hints-how-to-use-overload/) instead to narrow the types down depending on use.

Reply:
I agree, but that's a reflection of accepting a list, a dict, or a list of dicts in the same setter. `@overload` might help, but maybe a cleaner implementation would be using `singledispatchmethod`, but that's not available before python 3.8.